docs/user/TroubleShooting.md

```diff
@@ -0,0 +1,122 @@
+
+First, check if there are any pending CSRs:
+```sh
+oc get csr
+```
+If there are none, it means the kubelet did not start successfully.  If there is a pending CSR for the corresponding Machine/Node, then check the logs for the cluster-machine-approver; refer to the section "Important Pod Logs" above for exact steps.
+
+If the kubelet did not start successfully; the problem is related to either an invalid user-data secret (this is referenced from the Machine object) or some other problem with the ignition payload and/or the operating system.  In this case, you will need to consult the `machine-config-operator` documentation.
+
+## Machine Status: Phase Failed
+
+See the section **A Machine is listed as 'Failed'**
+below.
+
+# I deleted a Machine (or scaled down a MachineSet) but the Machine and/or Node did not go away
+
+This can be caused by a variety of reasons, such as invalid cloud credentials or PodDisruptionBudgets preventing the Node from draining.  The best place to look for information is the machine-controller's logs; refer to the section "Important Pod Logs" above for exact steps.
+
+# A Machine is listed as 'Failed'
+In this case, you'll need to take a look at the Machine's status and determine why the Machine entered a failed state.  In many instances, simply deleting the Machine object is sufficient.  In some other circumstances, the instance may need to be manually cleaned up directly from the cloud provider.  The best place to look for information is the ```machine-controller```'s logs; refer to the section [Important Pod Logs](#important-pod-logs) above for exact steps.
+
+If a Machine's status is failed, this means something unrecoverable has happened to the Machine.  It may be a Machine spec misconfiguration, the instance may have gone missing (eg, terminated by an outside actor) from the cloud.
+
+First, consult with the ```machine-controller```'s logs; refer to [Important Pod Logs](#important-pod-logs) above for exact steps.
+
+Next, compare your findings in the machine-controller logs with the cloud provider's configuration.
+
+You may need to restore proper credentials for the
+cloud provider, you may need to manually remove the
+instance from the cloud provider (if any), or you
+may simply need to correct this misconfiguration by
+adjusting the corresponding MachineSet or
+creating a new Machine object.  Finally, ensure
+you delete the corresponding Machine object.
+
+**IMPORTANT**
+
+Ensure you have reviewed and understand that
+Masters/Control Plane machines are not backed
+by MachineSets at the root of this document.
```

Comment:
i feel like we could shorten this and just provide a link back to the top, eg "Ensure you have reviewed <important info link> before proceeding"

Reply:
I thought about that, but I have trust issues.

Reply:
i can empathize, maybe leave the text **and** have a link back to the top lol